plugins/request/src/main/java/org/apache/unomi/plugins/request/actions/SetRemoteHostInfoAction.java

```diff
@@ -150,14 +148,19 @@ public int execute(Action action, Event event) {
             logger.error("Cannot lookup IP", e);
         }
 
-        UserAgentStringParser parser = UADetectorServiceFactory.getResourceModuleParser();
-        ReadableUserAgent agent = parser.parse(httpServletRequest.getHeader("User-Agent"));
-        session.setProperty("operatingSystemFamily", agent.getOperatingSystem().getFamilyName());
-        session.setProperty("operatingSystemName", agent.getOperatingSystem().getName());
-        session.setProperty("userAgentName", agent.getName());
-        session.setProperty("userAgentVersion", agent.getVersionNumber().toVersionString());
+        UserAgentAnalyzer uaa = UserAgentAnalyzer
```

Comment:
This should not be done for each rule execution as its creation is very expensive (multiple seconds). As documented here: https://yauaa.basjes.nl/README-Usage.html we should either use a shared instance that will have a synchronized call or use a ThreadPool or ThreadLocale of these agent analyzers.

Reply:
Sorry, I didn't check this part, I will make a proposal.

Reply:
@sergehuber @jbonofre : I made some tests and we can use a single shared instance. The first parsing is around ~2s and the next parsing is <1ms. 
I was thinking of creating an OSGi service dedicated to the UserAgentDetector and use it in the blueprint request plugin.
I also add a filter to only load the fields that we need.
I will make some load tests with JMeter to validate.
Thoughts?

Reply:
Sorry for the delay in answering. This sounds good ! Thanks

Reply:
Actually I was thinking instead of JMeter tests we could have a quick looping test in a unit test that would time the first init and then verify that subsequent calls are fast. Similar to what was done in the expression evaluators here : https://github.com/apache/unomi/blob/master/plugins/baseplugin/src/test/java/org/apache/unomi/plugins/baseplugin/conditions/PropertyConditionEvaluatorTest.java

Actually we also need to make those tests a little shorter and less aggressive because they are causing problems on some platforms (Windows)